packages/edit-site/src/components/global-styles/global-styles-provider.js

```diff
@@ -62,40 +69,74 @@ function useGlobalStylesUserConfig() {
 						_globalStylesId
 				  )
 				: undefined;
+			const _associatedStyleId = record
+				? record[ 'associated_style_id' ]
+				: undefined;
+			if ( _associatedStyleId ) {
+				getEditedEntityRecord(
+					'root',
+					'globalStyles',
+					_associatedStyleId
+				);
+			}
 
 			let hasResolved = false;
 			if (
+				! isReady &&
 				hasFinishedResolution(
 					'__experimentalGetCurrentGlobalStylesId'
 				)
 			) {
-				hasResolved = _globalStylesId
-					? hasFinishedResolution( 'getEditedEntityRecord', [
+				hasResolved = ( () => {
```

Comment:
I don't understand what `isReady` does that `hasResolved` doesn't? Is it a sort of `Promise.all()`?